safenode/src/network/event.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::{
+    error::{Error, Result},
+    safe_msg::SafeMsgCodec,
+    EventLoop, SafeRequest, SafeResponse,
+};
+use futures::{channel::oneshot, SinkExt};
+use libp2p::{
+    kad::{store::MemoryStore, GetProvidersOk, Kademlia, KademliaEvent, QueryResult},
+    mdns,
+    multiaddr::Protocol,
+    request_response::{self, ResponseChannel},
+    swarm::{NetworkBehaviour, SwarmEvent},
+};
+use tracing::{info, warn};
+
+#[derive(NetworkBehaviour)]
+#[behaviour(out_event = "SafeNodeEvent")]
+pub(super) struct SafeNodeBehaviour {
+    pub(super) request_response: request_response::Behaviour<SafeMsgCodec>,
+    pub(super) kademlia: Kademlia<MemoryStore>,
+    pub(super) mdns: mdns::async_io::Behaviour,
+}
+
+#[derive(Debug)]
+pub(super) enum SafeNodeEvent {
+    RequestResponse(request_response::Event<SafeRequest, SafeResponse>),
```

Comment:
request_response is actually not a very good name chosen by libp2p folks.
We should just map that to `Response` IMO. I mean, what other types of responses are there, other than to a `Request`? Unnecessary qualifier.

Reply:
Not sure if I understand this correctly, since we have `Event::Message::Request` and `Event::Message::Response` which denote an incoming request/response?